### PR TITLE
PE-739: Multi file upload

### PR DESCRIPTION
**Pull request by matibat**
Adds a flag for specifying multiple files to upload 
```shell
# Specifying multiple paths
$ yarn ardrive upload-file -w wallet.json -F "${PUBLIC_FOLDER_ID}" --local-paths ./package.json /another_file.txt ~/.bashrc

# Example for using GLOB
$ yarn ardrive upload-file -w wallet.json -F "${PUBLIC_FOLDER_ID}" --local-paths ./*.json
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -705,6 +707,17 @@ Users can perform a bulk upload by using the upload-file command on a target fol
 ardrive upload-file --local-file-path /path/to/folder  --parent-folder-id "9af694f6-4cfc-4eee-88a8-1b02704760c0" -w /path/to/wallet.json
 ```
 
+### Uploading files using a glob pattern (BETA)<a id="glob-upload"></a>
```
Comment (matibat) [inline README.md:710]: TODO: Rework this with the new specification

Comment (matibat) [reply README.md]: Let's see here: 7da6844

File: src/commands/tx_status.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ new CLICommand({
 		const txId = parameters.getRequiredParameterValue(TransactionIdParameter, TxID);
 		const transactionsInMempool = (await fetchMempool()).map((id) => new TransactionID(id));
 		const pending = transactionsInMempool.some((tx) => tx.equals(txId));
-		const confirmationAmount = confirmations ?? 15;
+		const confirmationAmount = +confirmations ?? 15;
```
Comment (arielmelendez) [inline src/commands/tx_status.ts:17]: Why was this change necessary?

File: src/parameter_declarations.ts [hunk 3]
```diff
@@ -325,3 +327,14 @@ Parameter.declare({
 		'(OPTIONAL) Derives a drive key for the given drive ID without verifying its correctness against the drive on chain.',
 	type: 'boolean'
 });
+
+Parameter.declare({
+	name: LocalPathsParameter,
+	aliases: ['--local-paths'],
+	type: 'array',
+	description: `(BETA) a glob pattern to select files to be uploaded
```
Comment (arielmelendez) [inline src/parameter_declarations.ts:335]: It's no longer a glob pattern, it's a list of files. We could add a note indicating that shell expansion is supported.

File: src/parameter_declarations.ts [hunk 4]
```diff
@@ -325,3 +327,14 @@ Parameter.declare({
...
+\t\t\t\t\t\t\t• Can NOT be used in conjunction with --local-files-path
+\t\t\t\t\t\t\t• Can NOT be used in conjunction with --local-files
+\t\t\t\t\t\t\t• Can lead to false positives or upload errors when glob string is not surrounded by quotation marks`,
+	forbiddenConjunctionParameters: [LocalFilesParameter, LocalFilePathParameter]
```
Comment (arielmelendez) [inline src/parameter_declarations.ts:339]: `DestinationFileNameParameter` should probably be disallowed as well, though one could argue that it's valid in the single file case.

File: src/commands/upload_file.ts [hunk 5]
```diff
@@ -69,14 +85,30 @@ new CLICommand({
...
 				});
+
 				return fileParameters;
 			}
 
+			const parentFolderId: FolderID = parameters.getRequiredParameterValue(ParentFolderIdParameter, EID);
+			const localFilesToUpload = parameters.getParameterValue<string[]>(LocalPathsParameter);
+
+			if (localFilesToUpload) {
+				const globParameters = localFilesToUpload.map((filePath: FilePath) => {
```
Comment (arielmelendez) [inline src/commands/upload_file.ts:96]: It's no longer a glob pattern.

File: src/CLICommand/parameters_helper.ts [hunk 6]
```diff
@@ -246,9 +246,13 @@ export class ParametersHelper {
 	 * @returns {string | undefined}
 	 * Returns the string value for the specific parameter; returns undefined if not set
 	 */
-	public getParameterValue(parameterName: ParameterName): string | undefined {
+	public getParameterValue<T = string>(
+		parameterName: ParameterName,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		mapFunc: (input: any) => T = (input: any) => input as T
+	): T {
```
Comment (arielmelendez) [inline src/CLICommand/parameters_helper.ts:253]: Does this still support potentially returning undefined after your changes? If not, is it causing crashes when optional parameters are being fetched this way?

**Review by fedellen — DISMISSED** [review]
👍

**Review by fedellen — APPROVED** [review]

**Review by jdaev — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Review by fedellen — APPROVED** [review]
